### ISSUE TITLE
Fails to build using qt5

### ISSUE THREAD

**Issue by matthijskooijman** [Embroidermodder/Embroidermodder#78]
I just tried compiling this on Debian Jessie, which uses qt5 by default. I ran into a few problems with that:
```
$ qmake
matthijs@grubby:~/docs/src/upstream/Embroidermodder/thumbnailer-kde4$ make
g++ -c -m64 -pipe -O2 -Wall -W -D_REENTRANT -fPIC -DQT_NO_DEBUG -DQT_PLUGIN -DQT_GUI_LIB -DQT_CORE_LIB -I/usr/lib/x86_64-linux-gnu/qt5/mkspecs/linux-g++-64 -I. -I. -isystem /usr/include/kde4 -I../libembroidery -isystem /usr/include/x86_64-linux-gnu/qt5 -isystem /usr/include/x86_64-linux-gnu/qt5/QtGui -isystem /usr/include/x86_64-linux-gnu/qt5/QtCore -I.moc -o .obj/libembroidery-thumbnailer-kde4.o libembroidery-thumbnailer-kde4.cpp
libembroidery-thumbnailer-kde4.cpp:2:26: fatal error: QGraphicsScene: No such file or directory
 #include <QGraphicsScene>
```
It seems QGraphicsScene moved from QtGui to QtWidgets in qt 5
```
embroidermodder2$ qmake
Project MESSAGE: debug build
Project ERROR: Unknown module(s) in QT: scripttools script
```
Not sure about this one.
```
$ qmake -v
QMake version 3.0
Using Qt version 5.3.2 in /usr/lib/x86_64-linux-gnu
```
Libembroidery compiled fine.

Using `qmake-qt4` to explicitly select q4 made all builds work fine. Perhaps the instructions should note this?

**Comment by redteam316** [comment]
@matthijskooijman,
There shouldn't be any major problem building embroidermodder2 with Qt5, although I don't think there was much testing done building the KDE4 thumbnailer with Qt5 because at the time we did most of the Qt5 porting, KDE4 didn't play nice with Qt5. I pushed a build fix for the thumbnailer here: 266f88e7e024e0b25f561a8a66e95144d8da69d0 

As far as the other issue involving scripttools, do this and let me know if it fixes the issue for you (it does for me):
[code]
sudo apt-get install qtscript5-dev qttools5-dev-tools
[/code]

Then doing:
[code]
qmake
make
[/code]
should work.

Ensuring you are building properly can be a little confusing at times if you have qt4 and qt5 on the same machine. As you note, explicitly using qmake-qt4 does work as expected, but since it should work with either qt4 or qt5, using qtchooser is likely preferable.

I.E.
[code]
qtchooser -qt=qt5 -run-tool=qmake
make
[/code]
or
[code]
qtchooser-qt=qt4 -run-tool=qmake
make
[/code]

Useful Packages:
[code]
qt4-default - Qt 4 development defaults package
qt5-default - Qt 5 development defaults package
qtchooser - Wrapper to select between Qt development binary versions
[/code]

The readme could use a bit of updating relating to this subject and since qt5 is the default now. Please let me know if I missed something in the explanation. If you want to send a pull request for the readme,  that is welcome since I don't have a fresh install of Jessie atm to check all the dependencies with.

**Comment by matthijskooijman** [comment]
With that commit and the extra packages you suggested, both the thumbnailer and embroidermodder compile with the default qt5 qmake as well now.

On my system, qmake is actually symlinked to qtchooser already, which defaults to qt5 now.

I don't have a fully clean Jessie system, but installing those packages (in addition to the ones in the README) was sufficient for me.
